### DISCUSSION TITLE
Example for `TextArea.code_editor` with syntax highlighting from the docs is not working

### DISCUSSION THREAD

**Discussion by kostrykin** [Textualize/textual#4911] [Q&A] [answered]
Trying this example, literally copy & paste: https://textual.textualize.io/widgets/text_area/#loading-text

```python
from textual.app import App, ComposeResult
from textual.widgets import TextArea

TEXT = """\
def hello(name):
    print("hello" + name)

def goodbye(name):
    print("goodbye" + name)
"""


class TextAreaExample(App):
    def compose(self) -> ComposeResult:
        yield TextArea.code_editor(TEXT, language="python")


app = TextAreaExample()
if __name__ == "__main__":
    app.run()
```

But the syntax highlighting isn't working:
<img width="360" alt="Bildschirmfoto 2024-08-21 um 10 45 55" src="https://github.com/user-attachments/assets/a39dc53f-561e-43ff-b9c4-a11a43e66e2f">

I have installed the `textual` conda package from https://anaconda.org/conda-forge/textual and I have also tried adding a `theme` parameter as suggested in https://github.com/Textualize/textual/discussions/4148.

Could it be that the `textual` conda package does not include the syntax highlighting capabilities? If so, what is the conda equivalent of `pip install "textual[syntax]"`?

**Comment by TomJGooding** [comment]
https://textual.textualize.io/widgets/text_area/#syntax-highlighting-dependencies

That `textual-textarea` package is a third-party library which predates the official Textual widget.

**Reply by kostrykin** [reply]
Oh, this is misleading, thanks for pointing out.

Does the `textual` conda package not include the syntax highlighting capabilities?

If so, what is the conda equivalent of `pip install "textual[syntax]"`?

**Reply by TomJGooding** [reply]
Sorry I'm not a conda user, but I was a bit surprised to discover that conda doesn't support optional dependencies! 

You might have to install these as separate packages?

**Reply by kostrykin** [reply, marked as answer]
Yes, however, it looks like there are no Conda packages for `textual[syntax]`.

I went this workaround, which works nicely. Inside my Conda environment file:
```yml
dependencies:
  - textual
  - pip
  - pip:
    - textual[syntax]
```
This way, `textual` is installed via pip, but its requirements are installed via Conda (at least those which do not come on top due to the syntax highlighting support).
